bin/migrate-oats-data/planning_review/decisions/init_planning_review_decisions.py

```diff
@@ -0,0 +1,188 @@
+                    )
+                except Exception as err:
+                    # this is NOT going to be caused by actual data insert failure. This code is only executed when the code error appears or connection to DB is lost
+                    logger.exception(err)
+                    conn.rollback()
+                    failed_inserts_count = count_total - successful_inserts_count
+                    last_planning_review_id = last_planning_review_id + 1
+
+    logger.info(
+        f"Finished {etl_name}: total amount of successful inserts {successful_inserts_count}, total failed inserts {failed_inserts_count}"
+    )
+
+
+def _insert_base_fields(conn, batch_size, cursor, rows):
+    parsed_data_list = _prepare_oats_planning_review_data(rows)
+
+    execute_batch(
+        cursor,
+        query,
+        parsed_data_list,
+        page_size=batch_size,
+    )
+
+    conn.commit()
+    return parsed_data_list
+
+
+query = f"""
+            INSERT INTO alcs.planning_review_decision (
+                planning_review_uuid,
+                decision_description,
+                audit_created_by,
+                was_released,
+                outcome_code,
+                resolution_number,
+                resolution_year,
+                date
+            )
+            VALUES (
+                %(review_uuid)s,
+                %(description)s,
+                '{OATS_ETL_USER}',
+                %(was_released)s,
+                %(outcome_code)s,
+                %(resolution_number)s,
+                %(resolution_year)s,
+                %(date)s
+            )
+"""
+
+
+def _prepare_oats_planning_review_data(row_data_list):
+    mapped_data_list = []
+    for row in row_data_list:
+        mapped_data_list.append(
+            {
+                "planning_decision_id": row["planning_decision_id"],
+                "review_uuid": row["uuid"],
+                "description": row["description"],
+                "date": _map_date(row),
+                "was_released": True,
+                "outcome_code": _map_outcome_code(row),
+                "resolution_number": row["resolution_number"],
+                "resolution_year": _map_resolution_year(row),
+            }
+        )
+
+    return mapped_data_list
+
+
+def _map_date(data):
+    date = data.get("decision_date", "")
+    d_date = add_timezone_and_keep_date_part(date)
+    return d_date
+
+
+def _map_outcome_code(data):
+    oats_code = data.get("planning_acceptance_code", "")
+    try:
+        alcs_name = OatsToAlcsDecisionOutcomes[oats_code].value.value
+        for outcome in AlcsPlanningReviewOutcomes:
+            if outcome.value == alcs_name:
+                return outcome.name
+    except KeyError:
+        file_number = data.get("planning_decision_id")
+        logger.info(
+            f"Key Error for{file_number}, no match to {oats_code} in ALCS, Override to OTHR"
+        )
+        return "OTHR"
+
```

Comment:
Instead of wrapping it into try except I would recommend checking if the key is present in Enum first and only then grab the corresponding value from Enum